### PR TITLE
Remove Unused List Types from the Slice Compiler

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
This is just a cleanup PR which:
- Removes some unused type-aliases we defined (for sets and lists)
- Removes the uncalled functions which returned them
- Updates some `for` loops to be `ranged-for` loops
- We used the identifier `enums` to refer to lists of enums and lists of enumerators...
   I fixed the enumerator lists to be called `enumerators`.

### CODE REVIEW

File: cpp/src/slice2php/Main.cpp [hunk 1]
```diff
@@ -889,20 +889,19 @@ CodeVisitor::visitEnum(const EnumPtr& p)
...
+            _out << nl << "const " << fixIdent(enumerator->name()) << " = " << enumerator->value() << ';';
         }
     }
 
     _out << eb;
 
     // Emit the type information.
     _out << sp << nl << type << " = IcePHP_defineEnum('" << scoped << "', array(";
-    for (EnumeratorList::iterator q = enums.begin(); q != enums.end(); ++q)
+    for (EnumeratorList::iterator q = enumerators.begin(); q != enumerators.end(); ++q)
```
Comment (pepone) [inline cpp/src/slice2php/Main.cpp:902]: Can we use a const iterator?

File: cpp/src/slice2py/PythonUtil.cpp [hunk 2]
```diff
@@ -1635,16 +1632,16 @@ Slice::Python::CodeVisitor::visitEnum(const EnumPtr& p)
...
-        string fixedEnum = fixIdent((*q)->name());
-        _out << nl << name << '.' << fixedEnum << " = " << name << "(\"" << (*q)->name() << "\", " << (*q)->value()
-             << ')';
+        string fixedEnum = fixIdent(enumerator->name());
+        _out << nl << name << '.' << fixedEnum << " = " << name << "(\"" << enumerator->name() << "\", "
+             << enumerator->value() << ')';
     }
     _out << nl << name << "._enumerators = { ";
-    for (q = enums.begin(); q != enums.end(); ++q)
+    for (EnumeratorList::iterator q = enumerators.begin(); q != enumerators.end(); ++q)
```
Comment (pepone) [inline cpp/src/slice2py/PythonUtil.cpp:1642]: Here too we should use const iterator if possible.

File: cpp/src/slice2rb/RubyUtil.cpp [hunk 3]
```diff
@@ -1018,7 +1018,7 @@ Slice::Ruby::CodeVisitor::visitEnum(const EnumPtr& p)
     //
     _out << sp;
     int i = 0;
-    for (EnumeratorList::iterator q = enums.begin(); q != enums.end(); ++q, ++i)
+    for (EnumeratorList::iterator q = enumerators.begin(); q != enumerators.end(); ++q, ++i)
```
Comment (pepone) [inline cpp/src/slice2rb/RubyUtil.cpp:1021]: ditto use const iterator

**Review by pepone — APPROVED** [review]

**Comment by InsertCreativityHere** [comment]
Replaced the `iterators` with `const_iterators`.
Makes you wonder why they weren't that way from the beginning...

**Pull request merged**